lib/tests/streamlit/runtime/pages_manager_test.py

```diff
@@ -88,11 +90,99 @@ def test_install_pages_watcher(self, patched_watch_dir):
         pages_manager.invalidate_pages_cache.assert_called_once()
 
 
+class PagesManagerV2Test(unittest.TestCase):
+    def setUp(self):
+        self.pages_manager = PagesManager("main_script_path")
+
+        # This signifies the change to V2
+        self.pages_manager.set_pages({})
+
+    def test_run_with_active_hash(self):
+        """Ensure the active script is set correctly"""
+        current_hash = self.pages_manager.main_script_hash
```

Comment:
nit: I feel like `main_script_hash` would be a better name here since the hash isn't always current throughout the test (it's not in the `run_with_active_hash` contextmanager, for example)